### ISSUE TITLE
Implement "export/import * from"

### ISSUE THREAD

**Issue by torch2424** [AssemblyScript/assemblyscript#27]
So after creating #21 

I now realize why I came to the conclusion import did not work.

If you try `export * from 'myfile'`, you will get the following console output:

<img width="638" alt="screen shot 2018-02-10 at 12 44 02 am" src="https://user-images.githubusercontent.com/1448289/36060305-afed46a6-0dfb-11e8-88e3-625fe026f3d1.png">

And if I remember correctly, imports dont work this way as well.

Thank you!

**Comment by dcodeIO** [comment]
That's right, wildcard exports and imports are not handled yet. Going to take a look and marking this issue as enhancement until then :)

**Comment by torch2424** [comment]
Awesome, thank you! :)

**Comment by gustavomick** [comment]
Any workaround in the meantime?

**Comment by dcodeIO** [comment]
Normal imports and exports should work, as should imports and exports of entire namespaces. Just the wildcard isn't implemented yet.

**Comment by gustavomick** [comment]
Thanks what if I need to import a wasm,eg. compiled  code from c++ to wasm then importing that to ts. Any viable way to make that work? Thanks.

**Comment by dcodeIO** [comment]
Linking wasms is a bit tricky currently. What I have in mind for this case is that the compiler could be provided with a base wasm, appending any additional code to it. would that work for your use case?

**Comment by gustavomick** [comment]
@dcodeIO thanks sorry about the late answer, i currently on vacations :). 

i was thinking about creating empty interfaces/type/mocks on typescript, using them from my main typescript code, then compiling to wasm and then replacing those mocks with real wasm implementations (that were previously compiled from c++ to wasm). that make sense? although not sure how to do that. if you have any suggestion will be helpful thanks.

**Comment by dcodeIO** [comment]
That should be possible as well. On the AssemblyScript side, you'd then have a .ts file that `declare`s the exported interface of the other wasm, which the AS module then imports. On the JS side, you'd then provide the other wasm for this import and it should just work.

```ts
declare namespace MyCppModule {
  export function someFunction(): i32;
}

// use it
MyCppModule.someFunction();
...
```

And simply give `{ MyCppModule: myCppModuleInstance }` as an import when instantiating the AS module.

**Comment by gustavomick** [comment]
@dcodeIO im missing something in here, how do i compile so my typescriptfile.wasm get linked to my cppfile.wasm file instead of the linked to the ts mock? 
i thought i would had to create a workaround script to merge them, replacing the mock with the real implementation, how that part should work?

**Comment by dcodeIO** [comment]
Merging, or real linking, might be more complicated because a linker will have to move around data segments and update their indexes where used. There is [wasm-merge](https://github.com/WebAssembly/binaryen#tools) which might, or might not, help.

My example above was based on the assumption that you'd then have two WebAssembly modules, with the exports of your cpp module being provided as imports to the typescript module.

**Comment by gustavomick** [comment]
ok ill try to read about this, thanks 👍

**Comment by torch2424** [comment]
Closed by #489 😄
